test/serial_test.py

```diff
@@ -0,0 +1,150 @@
+along with python-can. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import unittest
+import can
+from can.interfaces.serial.serial_can import SerialBus
+from mock import patch
+
+
+class SerialDummy:
+    """
+    Dummy to mock the serial communication
+    """
+    msg = None
+
+    def __init__(self):
+        self.msg = bytearray()
+
+    def read(self, size=1):
+        return_value = bytearray()
+        for i in range(size):
+            return_value.append(self.msg.pop(0))
+        return bytes(return_value)
+
+    def write(self, msg):
+        self.msg = bytearray(msg)
+
+    def reset(self):
+        self.msg = None
+
+
+class SimpleSerialTest(unittest.TestCase):
+
+    def setUp(self):
+        self.patcher = patch('serial.Serial')
+        self.mock_serial = self.patcher.start()
+        self.serial_dummy = SerialDummy()
+        self.mock_serial.return_value.write = self.serial_dummy.write
+        self.mock_serial.return_value.read = self.serial_dummy.read
+        self.addCleanup(self.patcher.stop)
+        self.bus = SerialBus('bus')
+
+    def tearDown(self):
+        self.serial_dummy.reset()
+
+    def test_rx_tx_min_max_data(self):
+        """
+        Tests the transfer from 0x00 to 0xFF for a 1 byte payload
+        """
+        for b in range(0, 255):
+            msg = can.Message(data=[b])
+            self.bus.send(msg)
+            msg_receive = self.bus.recv()
+            self.assertEqual(msg, msg_receive)
+
+    def test_rx_tx_min_max_dlc(self):
+        """
+        Tests the transfer from a 1 - 8 byte payload
+        """
+        payload = bytearray()
+        for b in range(1, 9):
+            payload.append(0)
+            msg = can.Message(data=payload)
+            self.bus.send(msg)
+            msg_receive = self.bus.recv()
+            self.assertEqual(msg, msg_receive)
+
+    def test_rx_tx_data_none(self):
+        """
+        Tests the transfer without payload
+        """
+        msg = can.Message(data=None)
+        self.bus.send(msg)
+        msg_receive = self.bus.recv()
+        self.assertEqual(msg, msg_receive)
+
+    def test_rx_tx_min_id(self):
+        """
+        Tests the transfer with the lowest arbitration id
+        """
+        msg = can.Message(arbitration_id=0)
+        self.bus.send(msg)
+        msg_receive = self.bus.recv()
+        self.assertEqual(msg, msg_receive)
+
+    def test_rx_tx_max_id(self):
+        """
+        Tests the transfer with the highest arbitration id
+        """
+        msg = can.Message(arbitration_id=536870911)
+        self.bus.send(msg)
+        msg_receive = self.bus.recv()
+        self.assertEqual(msg, msg_receive)
+
+    def test_rx_tx_max_timestamp(self):
+        """
+        Tests the transfer with the highest possible timestamp
+        """
+        msg = can.Message(timestamp=0xFFFFFFFF)
+        self.bus.send(msg)
+        msg_receive = self.bus.recv()
+        self.assertEqual(msg, msg_receive)
```

Comment:
When testing two messages if they are equal, timestamp is not compared. You need to test `msg.timestamp` and `msg_receive.timestamp` instead.